scripts/objectstore/copy_files_to_irods.py

```diff
@@ -0,0 +1,112 @@
+import os
+import uuid
+
+from psycopg2 import connect, sql
+
+from galaxy.util import directory_hash_id
+
+
+def copy_files_to_irods(start_dataset_id, end_dataset_id, object_store_info={"files1" : "/Users/kxk302/workspace/galaxy/database/files1", "files2" : "/Users/kxk302/workspace/galaxy/database/files2"}, irods_home="/tempZone/home/rods", connection_info={"dbname" : "galaxy", "user" : "postgres", "host" : "localhost", "password" : "password"}):
+    conn = None
+    cursor = None
+    sql_statement = None
+    sql_object = None
+    uuid_without_dash = None
+    uuid_with_dash = None
+    objectid = None
+    object_store_id = None
+    disk_sub_folder = None
+    disk_file_path = None
+    irods_sub_folder = None
+    irods_file_path = None
+    icommand_mkdir = None
+    icommand_put = None
+
+    if start_dataset_id > end_dataset_id:
+        print("start_dataset_id %d cannot be larger than end_dataset_id %d", start_dataset_id, end_dataset_id)
+        return
+
+    try:
+        # declare a new PostgreSQL connection object
+        conn = connect(
+            dbname=connection_info["dbname"],
+            user=connection_info["user"],
+            host=connection_info["host"],
+            password=connection_info["password"]
+        )
+        conn.cursor()
+
+    except Exception as e:
+        print(e)
+        return
+
+    osi_keys = list(object_store_info.keys())
+    osi_keys_quoted = ["'" + key + "'" for key in osi_keys]
```

Comment:
You actually don't need to worry about quoting these yourself, you can pass lists as arguments to a psycopg2 cursor's `execute()` method and it will properly quote for you. e.g.:

```python
sql = """SELECT id, object_store_id, uuid
                       FROM dataset
                       WHERE state = %s
                       AND NOT deleted
                       AND NOT purged
                       AND id >= %s
                       AND id <= %s
                       AND object_store_id IN %s
"""
args = ('ok', start_dataset_id, end_dataset_id, osi_keys)
print(cursor.mogrify(sql, args))
cursor.execute(sql, args)
```

Reply:
Revised this accordingly.